src/app/clusters/general-diagnostics-server/general-diagnostics-server.cpp

```diff
@@ -195,11 +195,23 @@ class GeneralDiagnosticsDelegate : public DeviceLayer::ConnectivityManagerDelega
     }
 
     // Gets called when the device has been rebooted.
-    void OnDeviceRebooted() override
+    void OnDeviceRebooted(uint8_t bootReason) override
     {
         ChipLogProgress(Zcl, "GeneralDiagnosticsDelegate: OnDeviceRebooted");
 
         ReportAttributeOnAllEndpoints(GeneralDiagnostics::Attributes::BootReasons::Id);
+
+        for (auto endpoint : EnabledEndpointsWithServerCluster(GeneralDiagnostics::Id))
+        {
+            // If GeneralDiagnostics cluster is implemented on this endpoint
```

Comment:
The comment seems outdated (now that we have `EnabledEndpointsWithServerCluster` helper).